### PR TITLE
runfix: Improve Typeahead removal behavior

### PR DESCRIPTION
**Pull request by atomrc**
## Description

There is a mess up when the Typeahead menu is being removed from the DOM. 
One of the scroll handler was not correctly removed which caused the element to be re-added to the DOM later on. 

This will make sure no listener are leftover when we unmount the Typeahead component

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx [hunk 1]
```diff
@@ -305,7 +305,7 @@ function useDynamicPositioning(
       return () => {
         resizeObserver.unobserve(targetElement);
         window.removeEventListener('resize', onReposition);
-        document.removeEventListener('scroll', handleScroll);
+        document.removeEventListener('scroll', handleScroll, {capture: true});
```
Comment (atomrc) [inline src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx:308]: TIL: You need to provide the same options provided to `addEventListener` for the removal to be actually done 🤦

File: src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx [hunk 2]
```diff
@@ -590,39 +603,13 @@ function useMenuAnchorRef(opt: UseMenuAnchorRefOptions): MutableRefObject<HTMLEl
...
-        if (rootElement !== null) {
-          rootElement.removeAttribute('aria-controls');
-        }
-
-        const containerDiv = anchorElementRef.current;
-        if (containerDiv !== null && containerDiv.isConnected) {
-          containerDiv.remove();
-        }
-      };
-    }
```
Comment (atomrc) [inline src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx:615]: This was never necessary with the way we use the plugin

File: src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx [hunk 3]
```diff
@@ -676,13 +663,6 @@ export function TypeaheadMenuPlugin<TOption extends TypeaheadOption>({
   const [editor] = useLexicalComposerContext();
   const [resolution, setResolution] = useState<Resolution | null>(null);
   const [menuVisible, setMenuVisible] = useState(false);
-  const anchorElementRef = useMenuAnchorRef({
-    containerId,
-    resolution,
-    setResolution,
-    className: `typeahead-menu ${anchorClassName || ''}`,
-    menuVisible,
-  });
```
Comment (atomrc) [inline src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx:685]: moving the anchor to a child component makes sure it's unmounted when there are no options to display

File: src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx [hunk 4]
```diff
@@ -748,13 +728,14 @@ export function TypeaheadMenuPlugin<TOption extends TypeaheadOption>({
     };
   }, [editor, triggerFn, onQueryChange, resolution, closeTypeahead, openTypeahead, menuVisible, setMenuVisible]);
 
-  return resolution === null || editor === null ? null : (
+  return resolution === null || editor === null || options.length === 0 ? null : (
```
Comment (atomrc) [inline src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx:731]: making sure we don't display the component if there are no options to show the user

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**